yt_dlp/extractor/rumble.py

```diff
@@ -77,12 +77,20 @@ def _real_extract(self, url):
                     formats.append(f)
         self._sort_formats(formats)
 
+        subtitles = {}
+        for lang, sub_info in video.get('cc', {}).items():
+            subtitles.setdefault(lang, []).append({
+                'url': sub_info['path'],
+                'name': sub_info.get('language', ''),
+            })
```

Comment:
```suggestion
        subtitles = {
            lang: [{
                'url': sub_info['path'],
                'name': sub_info.get('language') or '',
            }] for lang, sub_info in (video.get('cc') or {}).items() if sub_info.get('path')
        }
```

Reply:
Works too, I guess. Though admittedly Python syntax doesn’t make it look too appealing.

Reply:
On the other hand, the original version would need fewer changes if Rumble added streaming manifests with subtitle links. One could just move `subtitles = {}` earlier and `_merge_subtitles` into that, with the loop here unchanged.